### ISSUE TITLE
qt5 sub menu's don't work in wayland mode

### ISSUE THREAD

**Issue by unmellow** [swaywm/sway#5040]
* Sway Version: 1.4-5568f14e

Obtain a debug log like so:

    sway -d 2> ~/sway.log

This will record information about sway's activity when it's running. Briefly reproduce your problem and exit sway.  When preparing a debug log, brevity is important - start up sway, do the minimum work necessary to reproduce the error, then close sway.

Upload the debug log to a pastebin service such as [gist.github.com](https://gist.github.com), and link to it below.

* Debug Log: https://gist.github.com/unmellow/0afa06f1549079bd4569895fd6184281

You should try to reproduce the issue with the default configuration. If you cannot, please reproduce with a minimal configuration, upload the config to a pastebin service, and link to it below.

* Configuration File: https://gist.github.com/unmellow/0fa16db1441f1fbb2b122d9f254d643c

Finally, explain the steps you took in plain English to reproduce the problem below.

i explained it all here https://github.com/lxqt/pcmanfm-qt/issues/1083

**Comment by unmellow** [comment]
after restarting my computer i can no longer reproduce the bug maybe it's because i reinstalled dbus? idk

**Comment by unmellow** [comment]
i made a type when exported QT_QPA_PLARTFORM=wayland
the bug very much exist gosh i feel dumb

**Comment by emersion** [comment]
Can you provide a `WAYLAND_DEBUG=1` log of your client?

```sh
WAYLAND_DEBUG=1 your-client >client.log 2>&1
# Then reproduce the bug
```

**Comment by unmellow** [comment]
i think the issue is qt but i'll try it

**Comment by unmellow** [comment]
where does the log output?

**Comment by unmellow** [comment]
[pcmanfm-qt.log](https://github.com/swaywm/sway/files/4246283/pcmanfm-qt.log)
found it

**Comment by kennylevinsen** [comment]
```
[ 17251.421]  -> xdg_surface@22.set_window_geometry(0, 0, 959, 1050)
...
[ 17412.380]  -> xdg_positioner@28.set_anchor_rect(1914, 31, 1, 1)
[ 17412.405]  -> xdg_positioner@28.set_anchor(5)
[ 17412.415]  -> xdg_positioner@28.set_gravity(8)
[ 17412.424]  -> xdg_positioner@28.set_size(341, 308)
[ 17412.440]  -> xdg_surface@26.get_popup(new id xdg_popup@43, xdg_surface@22, xdg_positioner@28)
```

That anchor seems broken.

**Comment by jstraarup** [comment]
Sounds like a bug in Qt 5.14.1 - https://bugreports.qt.io/browse/QTBUG-81952

**Comment by mreppen** [comment]
I had similar issues with QT apps. The above posted bug report (https://bugreports.qt.io/browse/QTBUG-81952) lists it as fixed in 5.15.1. I installed 5.15.1 and menus appear to work.

The menus sometimes appear off screen, but that would maybe better be reported in its own issue?

**Comment by emersion** [comment]
>The menus sometimes appear off screen, but that would maybe better be reported in its own issue?

Yes.

**Comment by unmellow** [comment]
This bug has reappeared for me with 
qt version: 1.15.2
and sway: 1.5.1

**Comment by mreppen** [comment]
For what it's worth, it has not reappeared for me. I tried your file manager pcmanfm-qt and some other applications. I'm on Qt 5.15.2 and Sway 1.5.1. Maybe there is something else at play?

**Comment by unmellow** [comment]
Running sway in cage fixes my problem IDK what's going on
(I might be able to launch pcmanfm-qt in cage too)
I really have no idea should I make a new issue with
a new debug log?

**Comment by Soundtoxin** [comment]
I'm experiencing this issue on Sway with pcmanfm-qt as well. Menus don't appear unless I make it launch with xwayland.
